packages/components-native/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useRef } from "react";
+import { useIntl } from "react-intl";
+import { View } from "react-native";
+import { PrimaryAction, SecondaryAction } from "./ButtonGroupAction";
+import { messages } from "./messages";
+import { styles } from "./ButtonGroup.style";
+import { SecondaryActionSheet } from "./components/SecondaryActionSheet";
+import { getActions, usePreventTapWhenOffline } from "./utils";
+import { ButtonGroupActionElement } from "./types";
+import { Button } from "../Button";
+import { BottomSheetRef } from "../BottomSheet/BottomSheet";
+
+export interface ButtonGroupProps {
+  readonly children: ButtonGroupActionElement | ButtonGroupActionElement[];
+
+  /**
+   * Display a cancel button in the secondary bottom sheet footer.
+   */
+  readonly showCancelInBottomSheet?: boolean;
+
+  /**
+   * An optional heading to display in the secondary bottom sheet header.
+   */
+  readonly bottomSheetHeading?: string;
+
+  /**
+   * Callback that is called when the secondary actions bottom sheet is opened.
+   */
+  readonly onOpenBottomSheet?: () => void;
+
+  /**
+   * Callback that is called when the secondary actions bottom sheet is closed.
+   */
+  readonly onCloseBottomSheet?: () => void;
+
+  /**
+   * Allows you to Tap the button while offline
+   */
+  readonly allowTapWhenOffline?: boolean;
+}
+
+export function ButtonGroup({
+  children,
+  showCancelInBottomSheet,
+  bottomSheetHeading,
+  onOpenBottomSheet,
+  onCloseBottomSheet,
+  allowTapWhenOffline = false,
+}: ButtonGroupProps): JSX.Element {
+  const { formatMessage } = useIntl();
+  const { handlePress } = usePreventTapWhenOffline();
+  const secondaryActionsRef = useRef<BottomSheetRef>();
```

Comment:
Changed from `createRef` because it is recommended to `useRef` inside components because `createRef` remakes the ref every render